etherman/etherman.go

```diff
@@ -1295,18 +1296,29 @@ func decodeSequences(txData []byte, lastBatchNumber uint64, sequencer common.Add
 			return nil, err
 		}
 		coinbase := (data[1]).(common.Address)
+		dataAvailabilityMessage := (data[2]).([]byte) // TODO: is this right???
 		sequencedBatches := make([]SequencedBatch, len(sequencesValidium))
-		for i, seq := range sequencesValidium {
+		var batchNums []uint64
+		var hashes []common.Hash
+		for i := range sequencesValidium {
```

Comment:
Why don't capturing the current batch in the loop variable (instead of indexing `sequencesValidium` multiple times)?

Note that this needs further adjustments of the code, and I've put the suggestion just to illustrate my point. 🙂 

```suggestion
		for i, validiumData := range sequencesValidium {
```